src/eve/type_definitions.py

```diff
@@ -41,12 +42,37 @@
 )
 from pydantic.types import ConstrainedStr
 
-from .extended_typing import Any, Callable, Generator, Optional, Tuple, Type, Union
+from . import extended_typing as xtyping
+from .extended_typing import (
+    Any,
+    Callable,
+    Final,
+    Generator,
+    NoReturn,
+    Optional,
+    Tuple,
+    Type,
+    Union,
+    final,
+)
+
+
+frozenlist: Final = tuple
+frozendict: Final = _frozendict if sys.version_info >= (3, 9) else xtyping.FrozenDict
+
+
+@final
+class NothingType(type):
```

Comment:
General note:  While you are working on this, why not add docstrings at least to the new classes (and maybe one or two old ones per PR)? Currently the learning curve for anyone new in `eve` would be quite steep. And it makes it more likely for the reviews to be meaningful.

Reply:
Actually, I noticed after writing that that almost everything else had a doctoring. And if it becomes simpler then all the better.